src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -3,173 +3,94 @@
+                .url(url)
+                .build();
+        final Response response = newClient.newCall(request).execute();
+        try (ResponseBody responseBody = response.body()){
+
+            if (!response.isSuccessful()) {
+                final StringBuilder sb = new StringBuilder();
+                Headers requestHeaders = request.headers();
+                Headers responseHeaders = response.headers();
+
+                sb.append("{");
+                sb.append("\"Request\": {");
+                sb.append("\"URL\": \"").append(url).append("\",");
 
-            sb.append("\"URL\": \"").append(url).append("\",");
-            if (headers != null) {
                 sb.append("\"Headers\": {");
-                for (final Map.Entry<String, List<String>> header : headers.entrySet()) {
-                    final String key = header.getKey();
+                for (int i = 0, size = requestHeaders.size(); i < size; i++) {
+                    final String key = requestHeaders.name(i);
                     final String value;
                     if ("Authorization".equals(key)) {
                         value = "<Omitted>";
                     } else {
-                        value = String.join(",", header.getValue());
+                        value = String.join(",", requestHeaders.value(i));
                     }
                     sb.append("\"").append(key).append("\": \"").append(value).append("\",");
                 }
-            }
-            sb.append("}");
-
-            sb.append("}");
+                sb.append("}");
+                sb.append("}");
 
-            if (urlConnection != null) {
                 sb.append(", \"Response\": {");
-                for (final Map.Entry<String, List<String>> header : urlConnection.getHeaderFields().entrySet()) {
-                    final String key = header.getKey();
+                for (int i = 0, size = responseHeaders.size(); i < size; i++) {
+                    final String key = responseHeaders.name(i);
                     final String value;
                     if ("Set-Cookie".equals(key)) {
                         value = "<Omitted>";
                     } else {
-                        value = String.join(",", header.getValue());
+                        value = String.join(",", responseHeaders.value(i));
                     }
                     sb.append("\"").append(key).append("\": \"").append(value).append("\",");
                 }
-                sb.append("\"Code\": ").append(urlConnection.getResponseCode()).append(",");
-                sb.append("\"Message\": \"").append(urlConnection.getResponseMessage()).append("\",");
+                sb.append("\"Code\": ").append(response.code()).append(",");
+                sb.append("\"Message\": \"").append(response.message()).append("\",");
+                sb.append("\"Error Body\": \"").append(responseBody.string()).append("\"");
+                sb.append("}");
```

Comment:
Nit: Can replace `sb.append("}");` with `sb.append('}');` There's an IntelliJ inspection you can turn on to report these things; it claims there's a minor performance gain from not having to turn the character into a string.